code/datums/gamemodes/spy_theft.dm

```diff
@@ -552,46 +573,42 @@
 
 		active_bounties += B
 
-	//Add big station item bounties (copy paste. bad)
-	var/list/BS = big_station_bounties.Copy()
+	//Add big station item bounties
 	var/big_choice = null
 	var/difficulty = 0
 	var/obj/obj_existing = null
-	for(var/i=1, i<=big_station_bounty_amt, i++)
-		big_choice = null
-		obj_existing = null
-
-		//try to find an item that exists on the station zlevel
-		for(var/q=1, q<= 50, q++) //just like try 50 times i guess lol
-			LAGCHECK(LAG_LOW)
-			var/list/pair = pick(BS)
-			big_choice = pair[1]
-			difficulty = pair[2]
-			obj_existing = locate(big_choice)
-			if (obj_existing?.z == 1)
-				break
-			else
-				obj_existing = null
-
-		if (obj_existing?.z == 1)
-			var/datum/bounty_item/B = new /datum/bounty_item(src)
-			B.path = obj_existing.type
-			B.item = obj_existing
-			B.name = obj_existing.name
+	var/big_picked=1
+	while(big_picked<=big_station_bounty_amt)
+		if (big_station_bounties.len <= 0)
+			logTheThing( "debug", src, null, "spy_theft.dm was unable to create enough big station bounties." )
+			message_admins("Spy bounty logic was unable to create enough big station bounties.")
+			break
+		// Pick a known valid item, retrieve difficulty rating from other list
+		big_choice = pick(big_station_bounties)
+		obj_existing = pick(valid_spy_thief_targets_by_type[big_choice])
```

Comment:
You should check if this is non-empty and if it's empty remove it from the list and `continue` instead probably.

Reply:
I think I did what you were looking for.